adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/UserCredential.cs

```diff
@@ -26,50 +26,57 @@
 //------------------------------------------------------------------------------
 
 using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal;
+using Core = Microsoft.Identity.Core;
 
 namespace Microsoft.IdentityModel.Clients.ActiveDirectory
 {
-    internal enum UserAuthType
-    {
-        IntegratedAuth,
-        UsernamePassword
-    }
+    using UserAuthType = Core.UserAuthType;
+
 
     /// <summary>
     /// Credential used for integrated authentication on domain-joined machines.
     /// </summary>
     public class UserCredential
     {
+        internal Core.UserCredential _UserCredential;
```

Comment:
normally instance variables are not upper case, but use pascal casing.

Reply:
@brentschmaltz  By "not upper case, but pascal case", do you mean you want to see `userCredential`? (The term pascal case can mean opposite case, according to [Wikipedia entry Camel Case](https://en.wikipedia.org/wiki/Camel_case): "Some people and organizations, notably Microsoft, use the term camel case only for lower camel case. Pascal case means only upper camel case.")

In this particular case, the current class name is `UserCredential`, with a member which happens to be a similar type `Core.UserCredential`. The compiler does not like I name this member the same name `UserCredential` which conflicts with the class name. `userCredential` would work, though.